nixos/modules/services/desktops/pipewire.nix

```diff
@@ -25,17 +27,75 @@ in {
           Automatically run pipewire when connections are made to the pipewire socket.
         '';
       };
+
+      alsa = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Route audio to/from generic ALSA-using applications via the ALSA PIPEWIRE PCM plugin.
+          '';
+        };
+
+        support32Bit = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Whether to support sound for 32-bit ALSA applications on a 64-bit system.
+          '';
+        };
+      };
+
+      jack = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent JACK audio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
+
+      pulse = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent PulseAudio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
     };
   };
 
 
   ###### implementation
   config = mkIf cfg.enable {
-    environment.systemPackages = packages;
+    environment.systemPackages = [ pkgs.pipewire ];
 
-    systemd.packages = packages;
+    systemd.packages = [ pkgs.pipewire ];
 
     systemd.user.sockets.pipewire.wantedBy = lib.mkIf cfg.socketActivation [ "sockets.target" ];
-  };
 
+    sound.extraConfig = mkIf cfg.alsa.enable ''
+      pcm_type.pipewire {
+        libs.native = ${pkgs.pipewire.lib}/lib/alsa-lib/libasound_module_pcm_pipewire.so ;
+        ${optionalString enable32BitAlsaPlugins
+          "libs.32Bit = ${pkgs.pkgsi686Linux.pipewire.lib}/lib/alsa-lib/libasound_module_pcm_pipewire.so ;"}
```

Comment:
Would it make sense to add the paths to `passthru` inside pipewire so that we could add tests ensuring the files exist like we did [in fwupd](https://github.com/NixOS/nixpkgs/blob/8e2b14aceb1d40c7e8b84c03a7c78955359872bb/pkgs/os-specific/linux/firmware/fwupd/default.nix#L279-L304)?

Full-fledged tests would be even better but that is even more work.

Reply:
Good idea, I just added some tests for this.